consensus/cometbft/service/abci_utils.go

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2025, Berachain Foundation. All rights reserved.
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+//
+
+package cometbft
+
+import (
+	"strings"
+
+	errorsmod "cosmossdk.io/errors"
+	storetypes "cosmossdk.io/store/types"
+	abci "github.com/cometbft/cometbft/api/cometbft/abci/v1"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// NOTE: Partially copied from: https://github.com/cosmos/cosmos-sdk/blob/960d44842b9e313cbe762068a67a894ac82060ab/baseapp/abci.go#L1098
+func (s *Service) handleQueryStore(path []string, req *abci.QueryRequest) abci.QueryResponse {
+	queryable, ok := s.sm.GetCommitMultiStore().(storetypes.Queryable)
+	if !ok {
+		return queryResult(
+			errorsmod.Wrap(sdkerrors.ErrUnknownRequest, "multi-store does not support queries"),
+		)
+	}
+
+	req.Path = "/" + strings.Join(path[1:], "/") // req.Path == "/beacon"
+
+	if req.Height <= 1 && req.Prove {
```

Comment:
I understand this is copied from cosmos-sdk, but I wonder if this should be `s.initialHeight`?